### PR TITLE
Remove netstandard1.6 and netcoreapp1.1 builds

### LINKED ISSUE #725: Remove .NET Standard 1.6 build
.NET Standard 1.6 is no longer supported by the framework or Micrsoft - and is has various limitations in the engine. I propose removing it in the next release of the engine.

### PR DESCRIPTION
**Pull request by CharliePoole**
Fixes #725

### CODE REVIEW

**Comment by ChrisMaddock** [comment]
Thanks for doing this Charlie - I'd been meaning to get around to it for a while!

Do you mind if we put it on hold until after the 3.12 release? I'd prefer not to introduce any new major changes between the beta and the release, but I'm hoping to get the release out this weekend.

**Comment by CharliePoole** [comment]
No sweat! I'm working on some other stuff at the moment anyway.

**Comment by CharliePoole** [comment]
Rebased on latest master

File: build.cake [hunk 1]
```diff
@@ -48,6 +47,7 @@ var ZIP_IMG = PROJECT_DIR + "zip-image/";
 
 var SOLUTION_FILE = PROJECT_DIR + "NUnitConsole.sln";
 var ENGINE_CSPROJ = PROJECT_DIR + "src/NUnitEngine/nunit.engine/nunit.engine.csproj";
+var ENGINE_CORE_CSPROJ = PROJECT_DIR + "src/NUnitEngine/nunit.engine.core/nunit.engine.core.csproj";
```
Comment (mikkelbu) [inline build.cake:50]: This is not used?

File: src/NUnitEngine/EngineApiVersion.cs [hunk 2]
```diff
@@ -25,5 +25,5 @@
 
 [assembly: AssemblyProduct("NUnit Engine API")]
 [assembly: AssemblyVersion("3.0.0.0")]
-[assembly: AssemblyInformationalVersion("3.12.0")]
+[assembly: AssemblyInformationalVersion("3.12.0-beta1")]
```
Comment (mikkelbu) [inline src/NUnitEngine/EngineApiVersion.cs:28]: Should this be in another PR (if it should be part of the beta)?

File: src/NUnitEngine/mock-assembly/mock-assembly.csproj [hunk 3]
```diff
@@ -1,14 +1,11 @@
...
+    <TargetFrameworks>net35;netcoreapp2.1;netcoreapp3.1</TargetFrameworks>
     <SignAssembly>true</SignAssembly>
     <AssemblyOriginatorKeyFile>..\..\nunit.snk</AssemblyOriginatorKeyFile>
   </PropertyGroup>
-  <ItemGroup Condition="'$(TargetFramework)'=='netcoreapp1.1'">
-    <PackageReference Include="NUnit" Version="3.12.0" />
-  </ItemGroup>
-  <ItemGroup Condition="'$(TargetFramework)'!='netcoreapp1.1'">
+  <ItemGroup>
     <PackageReference Include="NUnit" Version="3.13.0-dev-06899" />
```
Comment (mikkelbu) [inline src/NUnitEngine/mock-assembly/mock-assembly.csproj:9]: This could be changed to the released version of NUnit 3.13 (but this should probably be in another PR)

**Review by mikkelbu — COMMENTED** [review]
All of the changes look sound, and I only have a couple of questions/comments. I tried to search for netstandard1.6 and netcoreapp1.1 and I think I found a couple of places that where missed:
* nunit-console\azure-pipelines.yml tries to publish netcoreapp1.1 test results which are missing (gives a warning on CI)
* nunit-console\src\NUnitEngine\nunit.engine.tests\Services\AgentProcessTests.cs has a commented TestCase mentioning netcoreapp1.1 - perhaps we should remove this
* nunit-console\nuget\engine\nunit.engine.nuspec lines 23-30 has a group targeting netstandard1.6 that can be removed

File: src/NUnitEngine/mock-assembly/mock-assembly.csproj (hunk 3 above)
Comment (CharliePoole) [reply src/NUnitEngine/mock-assembly/mock-assembly.csproj:9]: Yes, I agree.

**Comment by CharliePoole** [comment]
All good catches...  thanks.

**Comment by CharliePoole** [comment]
@mikkelbu I don't use Azure but I did the obvious-seeming changes. I'd appreciate it if you'd look at that file.

**Review by mikkelbu — APPROVED** [review]
LGTM and the changes to azure-pipelines.yml seem correct.

**Comment by CharliePoole** [comment]
Whoops! I didn't mean to merge this one - got confused with the equivalent on the GUI.

@ChrisMaddock Sorry... I know you wanted to hold on this till after the release. Please let me know what you want to do. I can back it out or you can cherry-pick or we can leave it in as you prefer.

**Comment by ChrisMaddock** [comment]
Hi Charlie,

I know we expect usage of .NET Standard 1.6 to be low-to-zero, but I'd still prefer we didn't merge such a major change in between the beta and the release of 3.12. I've undeleted the issue-725 branch and used GitHub's "revert" functionality to create #864. Would you mind recreated a new pull request from the issue-725 branch, to merge after release? Wouldn't want to put your work under my name! 😄 

Thanks for that. Planning to release 3.12 tomorrow, then we'll get this merged in.

**Comment by CharliePoole** [comment]
Yes... I restored the issue-725 branch but I'll wait to create a new pull request until after the release.
